src/rpc/resultcache.h

```diff
@@ -37,4 +48,23 @@ RPCResultCache& GetRPCResultCache();
 int GetLastValidatedHeight();
 void SetLastValidatedHeight(int height);
 
+struct CResultCache {
+    int height;
+    uint256 hash;
+    std::variant<CGetBurnInfoResult> data;
+};
+
+class LastResultCache {
```

Comment:
How about we call this `MemoizedResultCache` so it's obvious that's what it does.